### DISCUSSION TITLE
Minimum Distance Between Ellipse and Point

### DISCUSSION THREAD

**Discussion by VAK-53** [mpewsey/mpewsey.github.io#4] [General]
Matt, good for you for posting the actual article "Minimum Distance Between Ellipse and Point".
I still don't understand why the square of the distance is equal to (x1−x2)⋅(x1−x2). Imho, it is equal to the sum of squares of differences x1 and x2.

**Comment by mpewsey** [comment]
Just to clarify, that expression should be read as the dot product of those two vectors. In other words, (x1 - x2) dot (x1 - x2).

One way of writing the magnitude of a vector, v, is as sqrt(v dot v). Therefore, the square magnitude is v dot v.

I felt that using that particular form made the following evaluations easier, since you can work them in terms of vectors instead of algebraic components.

**Reply by VAK-53** [reply]
I get it now. I'll keep looking into it.
If you don't mind, I'll ask more questions.

**Reply by mpewsey** [reply]
Sure thing. If you have any other questions, please let me know.

**Reply by VAK-53** [reply]
Great article I've been looking for a long time. Its hallmark is its engineering approach.
I don't think there are any more questions about the theory. I will look into the programme.

**Reply by VAK-53** [reply]
![пояс](https://github.com/mpewsey/mpewsey.github.io/assets/66240976/19483955-131c-449a-8af9-6c3f304dca74)
Matt, I started experiments with my data: points are inside the ellipse near the boundary of the ellipse. 
I got the following results. I am quite satisfied with Newton's method, but I don't reject the gradient descent method yet, because it can be useful under certain conditions.
Why do you think method1 did not work on the given points?

**Reply by mpewsey** [reply]
The scale of your ellipse is much larger than the tests I did. So if the code you are using is indeed correct and saying that it is converging, then you might need to use a smaller learning rate to compensate for the scale.

What does the error vs. iteration plot look like? If it looks really noisy/spikey, then that could indicate a problem with the learning rate. If it's converging smoothly (to clearly incorrect solutions in this case), then that could be a problem with the code.

**Reply by VAK-53** [reply]
Matt,
my graph of error versus number of iterations looks very noisy/peaky. You write:
"The scale of your ellipse is much larger than the tests I did."
In that case, the gradient descent method probably won't work for me. 
The thing is, the sizes of the ellipses in my acoustic geometry problem can range from 20 metres to 20 kilometres.
On the other hand, I realised that I don't know the gradient descent method. I will study it.

**Reply by mpewsey** [reply]
If you select a smaller learning rate, you may still be able to get gradient descent to work for the scale of your problem. What I think may be happening is the learning rate is too high, so each iteration is advancing too far and skipping past the solution.

The default learning rate for the `find_distance1` method in the blog post is 0.01, which worked for ellipses less than 10 units in size. So perhaps try 0.01 / 1000 or 0.01 / 10000 since you are working on the order of thousands. Something like 0.01 / max(Ellipse Width, Ellipse Height) could also possibly work.

If adjusting the learning rate doesn't work, another approach would be to normalize your ellipse and points, solve for the the angles t1 and t2, and then use those to compute the distance at full scale.

**Reply by VAK-53** [reply]
Matt,
as we say, the last question in the row.
I am starting to insert your algorithm into my programme, after that I will write an article for Interneta "Construction of the hodograph of a sound wave point in an isotropic medium when the wind changes in height in the vertical plane. Return to geometric acoustics."
Can I refer to you in the article as the author of the algorithm?

**Reply by mpewsey** [reply]
If you want to include the blog post in your works cited, that's fine.

**Comment by VAK-53** [comment]
Matt,
tried learning_rate = 0.01 / 10000. The result is the same.
![шум](https://github.com/mpewsey/mpewsey.github.io/assets/66240976/ad1f2ef9-0bfb-4d2f-b6d2-f01fe709a3e6)
I suggest that we do not dwell on this story any longer.
Without really looking into the algorithm, I disabled the code to generate better initial angle based on vertices (# If point is inside ellipse).
For 15 points the time was reduced from 0.001532 to 0.000592 sec.

**Comment by VAK-53** [comment]
Mett,
I think there are some errors in the programme. 
In the get_point method:
        xe = 0.5 * self.width * math.cos(angle)
        ye = 0.5 * self.height * math.sin(angle)
and in the get_point:
        xe = 0.5 * self.width * np.cos(t)
        ye = 0.5 * self.height * np.sin(t)
If the angle/t is counted from the x-axis, you should change cos to sin, and vice versa.

**Reply by VAK-53** [reply]
Mett,
something's stuck in my head. I'm wrong. 
Everything in the programme is right.

**Reply by VAK-53** [reply]
It turns out that the angle fi in the parametric equation of the ellipse does not equal the angle of the vector-function of the ellipse.
The purpose of determining the angle of the vector-function is to construct a fragment of the ellipse.
